akka-typed/src/test/scala/akka/typed/TypedSpec.scala

```diff
@@ -60,8 +60,12 @@ class TypedSpec(val config: Config) extends TypedSpecSetup {
   implicit def scheduler = nativeSystem.scheduler
 
   override def afterAll(): Unit = {
-    Await.result(nativeSystem ? (Terminate(_)), timeout.duration): Status
-    Await.result(adaptedSystem ? (Terminate(_)), timeout.duration): Status
+    if (!nativeSystem.whenTerminated.isCompleted) {
+      Await.result(nativeSystem ? (Terminate(_)), timeout.duration): Status
+    }
+    if (!adaptedSystem.whenTerminated.isCompleted) {
+      Await.result(adaptedSystem ? (Terminate(_)), timeout.duration): Status
+    }
```

Comment:
Uff, the problem is that `ask` takes an implicit `scheduler`, rather than an execution context, which we can only get from an actor system, so the alternative would be to create an extra actor system just to shut down here. I can replace the asking with `terminate()` instead though.